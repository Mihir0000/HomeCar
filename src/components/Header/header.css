.header_section {
  width: 100%;
  float: left;
  background-image: url(../../../public/static/img/banner-bg.png);
  height: auto;
  background-size: 100%;
  padding: 10px 0px 25px 0px;
  background-repeat: no-repeat;
}

.logo {
  width: 42%;
  float: left;
}

.bg-transparent {
  background-color: transparent !important;
}

.navbar-expand-lg .navbar-nav .nav-link {
  padding-right: 20px;
  padding-left: 20px;
  color: #272a2a;
  font-size: 18px;
}
.navbar-light .navbar-nav .nav-link:focus,
.navbar-light .navbar-nav .nav-link:hover {
  color: #15e4d0;
}

.navbar {
  padding: 0.5rem 70px;
}
.navbar-light .navbar-nav .active > .nav-link,
.navbar-light .navbar-nav .nav-link.active,
.navbar-light .navbar-nav .nav-link.show,
.navbar-light .navbar-nav .show > .nav-link {
  color: #15e4d0;
}

button {
  border: 0;
  margin: 0;
  padding: 0;
  cursor: pointer;
}
.banner_section {
  width: 100%;
  float: left;
  padding-bottom: 105px;
}

.banner_taital {
  width: 100%;
  float: left;
  font-size: 60px;
  color: #15e4d0;
  font-weight: bold;
  line-height: 60px;
  padding-top: 120px;
}

.images_1 {
  width: 80%;
  float: right;
  padding-top: 50px;
}
.banner_text {
  width: 100%;
  float: left;
  font-size: 16px;
  margin: 0px;
  color: #252525;
}

.btn_main {
  width: 100%;
  float: left;
  margin-top: 30px;
}

.more_bt {
  width: 170px;
  float: left;
}

.more_bt a {
  width: 100%;
  float: left;
  font-size: 18px;
  color: #f6f8f8;
  background-color: #4bc5b8;
  text-align: center;
  padding: 10px 0px;
}

.contact_bt {
  width: 170px;
  float: left;
  margin-left: 15px;
}

.contact_bt a {
  width: 100%;
  float: left;
  font-size: 18px;
  color: #f6f8f8;
  background-color: #252525;
  text-align: center;
  padding: 10px 0px;
}

.contact_bt a:hover {
  color: #f6f8f8;
  background-color: #4bc5b8;
}


#main_slider .carousel-control-prev:hover,
#main_slider .carousel-control-next:hover,
#main_slider .carousel-control-prev:focus,
#main_slider .carousel-control-next:focus {
  border: 4px solid #4bc5b8;
  color: #4bc5b8;
}

#main_slider a.carousel-control-next {
  position: absolute;
  left: 200px;
  top: 85%;
}

#main_slider a.carousel-control-prev {
  position: absolute;
  left: 120px;
  top: 85%;
  border: 4px solid #4bc5b8;
  color: #4bc5b8;
}

#main_slider .carousel-control-prev,
#main_slider .carousel-control-next {
  width: 65px;
  height: 65px;
  border: 4px solid #000;
  opacity: 1;
  font-size: 30px;
  color: #000;
  border-radius: 100%;
}

#my_slider .carousel-control-prev:hover,
#my_slider .carousel-control-next:hover,
#my_slider .carousel-control-prev:focus,
#my_slider .carousel-control-next:focus {
  background-color: #19e9d9;
  color: #ffffff;
}

@media only screen and (min-width: 768px) and (max-width: 991px) {
  .header-search {
      padding: 15px 0px;
  }
}