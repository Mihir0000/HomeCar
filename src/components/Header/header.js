import React from 'react';
import { Col, Container, Nav, Row, Card, Image, Button } from 'react-bootstrap';
import './header.css';

const header = () => {
    const nav_toggler = () => {
        console.log('clicked');
    };

    return (
        <>
            <Container fluid className="header_section">
                <Container className="navbar navbar-expand-lg navbar-light bg-transparent">
                    <Card.Text className="logo bg-transparent">
                        <Card.Link href="/">
                            <Image src="static/img/logo.png" alt="logo_img" />
                        </Card.Link>
                    </Card.Text>
                    <Button
                        className="navbar-toggler"
                        type="button"
                        data-toggle="collapse"
                        data-target="#navbarSupportedContent"
                        aria-controls="navbarSupportedContent"
                        aria-expanded="false"
                        aria-label="Toggle navigation"
                    >
                        <Card.Subtitle
                            className="navbar-toggler-icon d-flex"
                            onClick={nav_toggler}
                        ></Card.Subtitle>
                    </Button>
                    <Container
                        className="collapse navbar-collapse justify-content-end"
                        id="navbarSupportedContent"
                    >
                        <Nav className="navbar-nav mr-auto">
                            <Nav.Item clasName="nav-item active">
                                <Nav.Link
                                    className="nav-link"
                                    href="index.html"
                                >
                                    Home
                                </Nav.Link>
                            </Nav.Item>
                            <Nav.Item className="nav-item">
                                <Nav.Link
                                    className="nav-link"
                                    href="health.html"
                                >
                                    Health
                                </Nav.Link>
                            </Nav.Item>
                            <Nav.Item className="nav-item">
                                <Nav.Link
                                    className="nav-link"
                                    href="medicine.html"
                                >
                                    Medicine
                                </Nav.Link>
                            </Nav.Item>
                            <Nav.Item className="nav-item">
                                <Nav.Link className="nav-link" href="news.html">
                                    News
                                </Nav.Link>
                            </Nav.Item>
                            <Nav.Item className="nav-item">
                                <Nav.Link
                                    className="nav-link"
                                    href="client.html"
                                >
                                    Client
                                </Nav.Link>
                            </Nav.Item>
                            <Nav.Item className="nav-item">
                                <Nav.Link
                                    className="nav-link"
                                    href="contact.html"
                                >
                                    Contact Us
                                </Nav.Link>
                            </Nav.Item>
                            <Nav.Item className="nav-item">
                                <Nav.Link className="nav-link" href="/">
                                    <Image
                                        src="static/img/search-icon.png"
                                        alt="search"
                                    />
                                </Nav.Link>
                            </Nav.Item>
                        </Nav>
                    </Container>
                </Container>

                <Container
                    fluid
                    id="main_slider"
                    className="carousel slide"
                    data-ride="carousel"
                >
                    <Container fluid className="carousel-inner">
                        <Card className="carousel-item active bg-transparent border-0">
                            <Card.Body className="banner_section">
                                <Container>
                                    <Row>
                                        <Col md={6}>
                                            <Card.Title className="banner_taital fs-1">
                                                <Card.Subtitle className="fs-1">
                                                    Health
                                                </Card.Subtitle>
                                                <Card.Subtitle
                                                    className="fs-1"
                                                    style={{ color: '#151515' }}
                                                >
                                                    Care
                                                </Card.Subtitle>
                                            </Card.Title>
                                            <Card.Text className="banner_text mb-3">
                                                There are many variations of
                                                passages of Lorem Ipsum
                                            </Card.Text>
                                            <Card.Subtitle className="btn_main">
                                                <Card.Subtitle className="more_bt">
                                                    <Card.Link href="/">
                                                        Contact Now
                                                    </Card.Link>
                                                </Card.Subtitle>
                                                <Card.Subtitle className="contact_bt">
                                                    <Card.Link href="/">
                                                        Get A Quote
                                                    </Card.Link>
                                                </Card.Subtitle>
                                            </Card.Subtitle>
                                        </Col>
                                        <Col md={6}>
                                            <Card.Subtitle className="image_1">
                                                <Image
                                                    src="static/img/img-1.png"
                                                    alt="img1"
                                                />
                                            </Card.Subtitle>
                                        </Col>
                                    </Row>
                                </Container>
                            </Card.Body>
                        </Card>
                        {/* <div className="carousel-item">
                            <div className="banner_section">
                                <div className="container">
                                    <div className="row">
                                        <div className="col-md-6">
                                            <h1 className="banner_taital">
                                                Health <br />
                                                <span
                                                    style={{ color: '#151515' }}
                                                >
                                                    Care
                                                </span>
                                            </h1>
                                            <p className="banner_text">
                                                There are many variations of
                                                passages of Lorem Ipsum
                                            </p>
                                            <div className="btn_main">
                                                <div className="more_bt">
                                                    <a href="/">Contact Now</a>
                                                </div>
                                                <div className="contact_bt">
                                                    <a href="/">Get A Quote</a>
                                                </div>
                                            </div>
                                        </div>
                                        <div className="col-md-6">
                                            <div className="image_1">
                                                <img
                                                    src="static/img/img-1.png"
                                                    alt="img1"
                                                />
                                            </div>
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </div>
                        <div className="carousel-item">
                            <div className="banner_section">
                                <div className="container">
                                    <div className="row">
                                        <div className="col-md-6">
                                            <h1 className="banner_taital">
                                                Health <br />
                                                <span
                                                    style={{ color: '#151515' }}
                                                >
                                                    Care
                                                </span>
                                            </h1>
                                            <p className="banner_text">
                                                There are many variations of
                                                passages of Lorem Ipsum
                                            </p>
                                            <div className="btn_main">
                                                <div className="more_bt">
                                                    <a href="/">Contact Now</a>
                                                </div>
                                                <div className="contact_bt">
                                                    <a href="/">Get A Quote</a>
                                                </div>
                                            </div>
                                        </div>
                                        <div className="col-md-6">
                                            <div className="image_1">
                                                <img
                                                    src="static/img/img-1.png"
                                                    alt="img1"
                                                />
                                            </div>
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </div> */}
                    </Container>
                    <Card.Link
                        className="carousel-control-prev"
                        href="#main_slider"
                        role="button"
                        data-slide="prev"
                    >
                        <i className="fa fa-long-arrow-left"></i>
                    </Card.Link>
                    <Card.Link
                        className="carousel-control-next"
                        href="#main_slider"
                        role="button"
                        data-slide="next"
                        style={{ fontSize: '24px', paddingTop: '4px' }}
                    >
                        <i
                            className="fa fa-long-arrow-right"
                            style={{ fontSize: '24px', paddingTop: '4px' }}
                        ></i>
                    </Card.Link>
                </Container>
            </Container>
        </>
    );
};

export default header;
