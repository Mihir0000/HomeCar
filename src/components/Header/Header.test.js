import React from 'react';
import { render, screen } from '@testing-library/react';
import Header from './header';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

test('left right by role', () => {
    render(<Header />);
    const leftRightElement = screen.getByRole('contact_btn');
    expect(leftRightElement).toBeInTheDocument();
});

test('Button should be enabled', () => {
    render(<Header />);
    const inputBtn = screen.getByTestId('enable_btn');
    expect(inputBtn).not.toBeDisabled;
});

it('snapshot testing --header', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
});

// crash test --header
it('render without crashing --header', () => {
    const div = document.createElement('div');
    render(<Header />, div);
    ReactDOM.unmountComponentAtNode(div);
});
