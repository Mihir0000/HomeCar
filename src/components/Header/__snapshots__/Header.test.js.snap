// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing --header 1`] = `
<Fragment>
  <Container
    className="header_section"
    fluid={true}
  >
    <Container
      className="navbar navbar-expand-lg navbar-light bg-transparent"
      fluid={false}
    >
      <CardText
        className="logo bg-transparent"
      >
        <CardLink
          href="/"
        >
          <Image
            alt="logo_img"
            fluid={false}
            rounded={false}
            roundedCircle={false}
            src="static/img/logo.png"
            thumbnail={false}
          />
        </CardLink>
      </CardText>
      <Button
        active={false}
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-target="#navbarSupportedContent"
        data-testid="enable_btn"
        data-toggle="collapse"
        disabled={false}
        type="button"
        variant="primary"
      >
        <CardSubtitle
          className="navbar-toggler-icon d-flex"
        />
      </Button>
      <Container
        className="collapse navbar-collapse justify-content-end"
        fluid={false}
        id="navbarSupportedContent"
      >
        <Nav
          className="navbar-nav mr-auto"
          fill={false}
          justify={false}
        >
          <NavItem
            className="nav-item active"
          >
            <NavLink
              className="nav-link"
              disabled={false}
              href="index.html"
            >
              Home
            </NavLink>
          </NavItem>
          <NavItem
            className="nav-item"
          >
            <NavLink
              className="nav-link"
              disabled={false}
              href="health.html"
            >
              Health
            </NavLink>
          </NavItem>
          <NavItem
            className="nav-item"
          >
            <NavLink
              className="nav-link"
              disabled={false}
              href="medicine.html"
            >
              Medicine
            </NavLink>
          </NavItem>
          <NavItem
            className="nav-item"
          >
            <NavLink
              className="nav-link"
              disabled={false}
              href="news.html"
            >
              News
            </NavLink>
          </NavItem>
          <NavItem
            className="nav-item"
          >
            <NavLink
              className="nav-link"
              disabled={false}
              href="client.html"
            >
              Client
            </NavLink>
          </NavItem>
          <NavItem
            className="nav-item"
          >
            <NavLink
              className="nav-link"
              disabled={false}
              href="contact.html"
            >
              Contact Us
            </NavLink>
          </NavItem>
          <NavItem
            className="nav-item"
          >
            <NavLink
              className="nav-link"
              disabled={false}
              href="/"
            >
              <Image
                alt="search"
                fluid={false}
                rounded={false}
                roundedCircle={false}
                src="static/img/search-icon.png"
                thumbnail={false}
              />
            </NavLink>
          </NavItem>
        </Nav>
      </Container>
    </Container>
    <Container
      className="carousel slide"
      data-ride="carousel"
      fluid={true}
      id="main_slider"
    >
      <Container
        className="carousel-inner"
        fluid={true}
      >
        <Card
          body={false}
          className="carousel-item active bg-transparent border-0"
        >
          <CardBody
            className="banner_section"
          >
            <Container
              fluid={false}
            >
              <Row>
                <Col
                  md={6}
                >
                  <CardTitle
                    className="banner_taital fs-1"
                  >
                    <CardSubtitle
                      className="fs-1"
                    >
                      Health
                    </CardSubtitle>
                    <CardSubtitle
                      className="fs-1"
                      style={
                        Object {
                          "color": "#151515",
                        }
                      }
                    >
                      Care
                    </CardSubtitle>
                  </CardTitle>
                  <CardText
                    className="banner_text mb-3"
                  >
                    There are many variations of passages of Lorem Ipsum
                  </CardText>
                  <CardSubtitle
                    className="btn_main"
                  >
                    <CardSubtitle
                      className="more_bt"
                    >
                      <CardLink
                        href="/"
                        role="contact_btn"
                      >
                        Contact Now
                      </CardLink>
                    </CardSubtitle>
                    <CardSubtitle
                      className="contact_bt"
                    >
                      <CardLink
                        href="/"
                      >
                        Get A Quote
                      </CardLink>
                    </CardSubtitle>
                  </CardSubtitle>
                </Col>
                <Col
                  md={6}
                >
                  <CardSubtitle
                    className="image_1"
                  >
                    <Image
                      alt="img1"
                      fluid={false}
                      rounded={false}
                      roundedCircle={false}
                      src="static/img/img-1.png"
                      thumbnail={false}
                    />
                  </CardSubtitle>
                </Col>
              </Row>
            </Container>
          </CardBody>
        </Card>
      </Container>
      <CardLink
        className="carousel-control-prev"
        data-slide="prev"
        href="#main_slider"
        role="button"
      >
        <i
          className="fa fa-long-arrow-left"
        />
      </CardLink>
      <CardLink
        className="carousel-control-next"
        data-slide="next"
        href="#main_slider"
        role="button"
        style={
          Object {
            "fontSize": "24px",
            "paddingTop": "4px",
          }
        }
      >
        <i
          className="fa fa-long-arrow-right"
          style={
            Object {
              "fontSize": "24px",
              "paddingTop": "4px",
            }
          }
        />
      </CardLink>
    </Container>
  </Container>
</Fragment>
`;
